front/src/applications/stdcm/views/OSRDSTDCM.tsx

```diff
@@ -1,27 +1,21 @@
 import React, { useEffect, useState } from 'react';
-import { useDispatch, useSelector } from 'react-redux';
-import { getMode } from 'reducers/osrdconf/selectors';
-import { updateSelectedTrainId, updateSelectedProjection } from 'reducers/osrdsimulation/actions';
-import { MODES } from 'applications/operationalStudies/consts';
+import { useDispatch } from 'react-redux';
+
 import STDCM_REQUEST_STATUS from 'applications/stdcm/consts';
-import { StdcmRequestStatus } from 'applications/stdcm/types';
-import { updateMode } from 'reducers/osrdconf';
-import OSRDStdcmConfig from './OSRDStdcmConfig';
-import StdcmRequestModal from './StdcmRequestModal';
+import type { StdcmRequestStatus } from 'applications/stdcm/types';
+import OSRDStdcmConfig from 'applications/stdcm/views/OSRDStdcmConfig';
+import StdcmRequestModal from 'applications/stdcm/views/StdcmRequestModal';
+
+import { updateSelectedTrainId, updateSelectedProjection } from 'reducers/osrdsimulation/actions';
 
 export default function OSRDSTDCM() {
   const dispatch = useDispatch();
-  const mode = useSelector(getMode);
   const [currentStdcmRequestStatus, setCurrentStdcmRequestStatus] = useState<StdcmRequestStatus>(
     STDCM_REQUEST_STATUS.idle
   );
   useEffect(() => {
-    if (mode !== MODES.stdcm) dispatch(updateMode(MODES.stdcm));
-    return () => {
-      dispatch(updateMode(MODES.simulation));
-      dispatch(updateSelectedTrainId(undefined));
-      dispatch(updateSelectedProjection(undefined));
-    };
+    dispatch(updateSelectedTrainId(undefined));
+    dispatch(updateSelectedProjection(undefined));
```

Comment:
I think there is a mistake here, the behaviour is not the same ?
```suggestion
    () => {
      dispatch(updateSelectedTrainId(undefined));
      dispatch(updateSelectedProjection(undefined));
    }
```